### PR TITLE
Step reader - handling inconsistencies from step files

### PR DESCRIPTION
**Pull request by GabrielJMS**
* **Please check if the PR fulfills these requirements**
- [x] The commit messages follows our guidelines (explicit, in english)
- [x] Tests for the changes have been added (for bug fixes / features)
tested with door step file
- [x] Docs have been added / updated (for bug fixes / features)
- [x] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [ ] Bug fix:
- [x] New features:
- [ ] Documentation
- [ ] Optimization

* **Other information**:

image
image
image
image
image
image
image
image (10)
SOLID_COMPARAISON

### CODE REVIEW

File: volmdlr/faces.py [hunk 1]
```diff
@@ -4007,13 +4004,16 @@ def bsplinecurve3d_to_2d(self, bspline_curve3d):
...
-                                            bspline_curve3d.points[-1])
-        flag = True
-        for pt in bspline_curve3d.points:
-            if not bsc_linesegment.point_belongs(pt):
-                flag = False
-                break
+        start = bspline_curve3d.points[0]
+        end = bspline_curve3d.points[-1]
+        flag = False
+        if start != end:
```
Comment (WirajanDASILVA) [inline volmdlr/faces.py:4010]:
```suggestion
        if not start.is_close(end):
```

**Review by WirajanDASILVA — APPROVED** [review]

File: volmdlr/faces.py [hunk 2]
```diff
@@ -3491,7 +3489,7 @@ def point3d_to_2d(self, point3d):
         """
         Transform a 3D Cartesian point (x, y, z) into a parametric (u, v) point.
         """
-        x, y, _ = self.frame.global_to_local_coordinates(point3d)
+        x, y, z = self.frame.global_to_local_coordinates(point3d)
```
Comment (masfaraud) [inline volmdlr/faces.py:3492]: is z used? _ was there to avoid unused variable pylint error

File: volmdlr/step.py [hunk 3]
```diff
@@ -870,32 +891,42 @@ def parse_arguments(self, arguments):
                 argument.append(arg_id)
                 arguments[i] = argument
 
-    def instantiate(self, name, arguments, object_dict):
+    def instanciate(self, name, arguments, object_dict, step_id):
```
Comment (masfaraud) [inline volmdlr/step.py:894]: I changed on dev instanciate to instantiate (correct the typo). Suggest to rename to instantiate

File: volmdlr/step.py [hunk 4]
```diff
@@ -968,9 +1000,9 @@ def to_volume_model(self, show_times: bool = False):
                     for instanciate_id in instanciate_ids[::-1]:
                         t = time.time()
                         arguments = self.functions[instanciate_id].arg[:]
-                        volmdlr_object = self.instantiate(
```
Comment (masfaraud) [inline volmdlr/step.py:971]: instantiate

File: volmdlr/step.py [hunk 5]
```diff
@@ -1017,7 +1060,7 @@ def to_points(self):
         points3d = []
         for stepfunction in self.functions.values():
             if stepfunction.name == 'CARTESIAN_POINT':
-                # INSTANTIATION
+                # INSTANCIATION
```
Comment (masfaraud) [inline volmdlr/step.py:1063]: typo

**Review by masfaraud — CHANGES_REQUESTED** [review]

**Review by masfaraud — APPROVED** [review]

**Pull request merged**